src/types.rs

```diff
@@ -370,9 +403,333 @@ impl From<ReturnCode> for u8 {
     }
 }
 
+/// RPC payload
+pub type RpcPayload<'a> = &'a [u8];
+
+#[derive(Debug, PartialEq)]
+/// SD payload
+pub struct SdPayload {
+    /// Flags
+    pub flags: SdFlags,
+    /// Entries
+    pub entries: Vec<SdEntry>,
+    /// Options
+    pub options: Vec<SdOption>,
+}
+
+impl SdPayload {
+    /// Length of the payload in bytes
+    pub(crate) fn len(&self) -> usize {
+        let mut len: usize = SD_HEADER_LENGTH;
+
+        len += 4;
+        len += self.entries_len();
+
+        len += 4;
+        len += self.options_len();
+
+        len
+    }
```

Comment:
Would be interesting how the compiler can optimise this. Does it make a difference between this version and

```rust
fn len(&self) -> usize {
  SD_HEADER_LENGTH + 4 + self.entries.len() + 4 + self.options.len()
}
```

Reply:
done.